glib/src/auto/functions.rs

```diff
@@ -571,16 +571,15 @@ pub fn get_application_name() -> Option<GString> {
     unsafe { from_glib_none(glib_sys::g_get_application_name()) }
 }
 
-pub fn get_charset() -> Option<GString> {
-    unsafe {
-        let mut charset = ptr::null();
-        let ret = from_glib(glib_sys::g_get_charset(&mut charset));
-        if ret {
-            Some(from_glib_none(charset))
-        } else {
-            None
-        }
-    }
+/// Obtain the character set for the current locale.
+///
+/// This retuns whether the locale's encoding is UTF-8, and the current
+/// charset if aavailable.
+pub fn get_charset() -> (bool, Option<GString>) {
+    let mut charset = ptr::null();
+    let is_utf8 = unsafe { from_glib(glib_sys::g_get_charset(&mut charset)) };
+    let charset = unsafe { from_glib_none(charset) };
```

Comment:
Wrap the whole block into a single `unsafe`. They're all conceptually the same unsafe block and the invariants that have to hold are now spanning multiple unsafe blocks. That's not good.

They should be kept as small as possible, but everything before/after the unsafe block should have normal Rust safety rules again so that everything is nicely wrapped up inside the unsafe block.